sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedFlux.java

```diff
@@ -17,6 +20,29 @@
  * @see ContinuablePage
  */
 public abstract class ContinuablePagedFlux<C, T, P extends ContinuablePage<C, T>> extends Flux<T> {
+    final Predicate<C> continuationPredicate;
+
+    /**
+     * Creates an instance of ContinuablePagedFlux.
+     * <p>
+     * Continuation completes when the last returned continuation token is null.
+     */
+    protected ContinuablePagedFlux() {
+        this(Objects::nonNull);
+    }
+
+    /**
+     * Creates an instance of ContinuablePagedFlux.
+     * <p>
+     * If {@code continuationPredicate} is null then the predicate will only check if the continuation token is
+     * non-null.
+     *
+     * @param continuationPredicate A predicate which determines if paging should continue.
+     */
+    protected ContinuablePagedFlux(Predicate<C> continuationPredicate) {
+        this.continuationPredicate = (continuationPredicate == null) ? Objects::nonNull : continuationPredicate;
```

Comment:
Since `continuationPredicate` is package-private, this will not be accessible to any sub-classes outside of this package.